cyclonedx-bom/src/external_models/uri.rs

```diff
@@ -105,45 +98,40 @@ pub enum UriError {
 mod test {
     use pretty_assertions::assert_eq;
 
-    use super::*;
+    use crate::{
+        external_models::uri::{validate_purl, validate_uri},
+        prelude::{Purl, Uri},
+    };
 
     #[test]
     fn valid_purls_should_pass_validation() {
-        let validation_result = Purl("pkg:cargo/cyclonedx-bom@0.3.1".to_string()).validate();
+        let validation_result = validate_purl(&Purl("pkg:cargo/cyclonedx-bom@0.3.1".to_string()));
 
-        assert_eq!(validation_result, ValidationResult::Passed);
+        assert_eq!(Ok(()), validation_result);
```

Comment:
`is_ok()` would not provide an informative error message - it would just report `false` without debug-printing the `Failed` variant with its associated data.